src/infrastructure/implementations/blockstack-crux-user-repository.ts

```diff
@@ -0,0 +1,117 @@
+            infrastructure.gaiaHub = gaiaHubOverride || infrastructure.gaiaHub;
+            const subdomainRegistrarOverride = options.cruxDomain.config.nameserviceConfiguration ? options.cruxDomain.config.nameserviceConfiguration.subdomainRegistrar : undefined;
+            infrastructure.subdomainRegistrar = subdomainRegistrarOverride || infrastructure.subdomainRegistrar;
+        }
+        this.infrastructure = infrastructure;
+        this.blockstackService = new BlockstackService({
+            bnsNodes: this.infrastructure.bnsNodes,
+            cacheStorage: this.cacheStorage,
+            subdomainRegistrar: this.infrastructure.subdomainRegistrar,
+        });
+        log.debug("BlockstackCruxUserRepository initialised");
+    }
+    public create = async (cruxId: CruxId, keyManager: IKeyManager): Promise<CruxUser> => {
+        const cruxUserInformation = await this.blockstackService.registerCruxId(cruxId, this.infrastructure.gaiaHub, keyManager);
+        return new CruxUser(cruxId, {}, cruxUserInformation);
+    }
+    public isCruxIdAvailable = async (cruxID: CruxId): Promise<boolean> => {
+        return this.blockstackService.isCruxIdAvailable(cruxID);
+    }
+    public getByCruxId = async (cruxID: CruxId, tag?: string, onlyRegistered: boolean = true): Promise<CruxUser|undefined> => {
+        const cruxUserInformation = await this.blockstackService.getCruxIdInformation(cruxID);
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.NONE) {
+            return;
+        }
+        let addressMap = {};
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.DONE) {
+            addressMap = await this.getAddressMap(cruxID, tag);
+        } else {
+            if (onlyRegistered) {
+                return;
+            }
+        }
+        return new CruxUser(cruxID, addressMap, cruxUserInformation);
+    }
+    public getWithKey = async (keyManager: IKeyManager, cruxDomainId: CruxDomainId): Promise<CruxUser|undefined> => {
+        const cruxID = await this.blockstackService.getCruxIdWithKeyManager(keyManager, cruxDomainId);
+        if (!cruxID) {
+            return;
+        }
+        const cruxUserInformation = await this.blockstackService.getCruxIdInformation(cruxID);
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.NONE) {
+            return;
+        }
+        let addressMap = {};
+        if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.DONE) {
+            addressMap = await this.getAddressMap(cruxID);
+        } else if (cruxUserInformation.registrationStatus.status === SubdomainRegistrationStatus.PENDING) {
+            addressMap = await this.getAddressMap(cruxID, publicKeyToAddress(await keyManager.getPubKey()));
```

Comment:
I am not sure but will it not map to tag field instead of ownerAddress field in getAddressMap method. 

can we be more explicit here to avoid any confusion?

Reply:
fixed